src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -791,6 +791,9 @@ private static void LoadDataSources(CanvasDocument app, DirectoryReader director
                     localDatabaseReferenceJson.dataSources.Add(tableDef.EntityName, tableDef.LocalReferenceDSJson);
                 }
             }
+            //Skip the data source references with empty data sources. This avoids data source reference with empty json for data sources
+            app._dataSourceReferences = dataSourceReferences.Where(dataSourceReferenceEntry => dataSourceReferenceEntry.Value.dataSources.Count > 0)
+                                                            .ToDictionary(dataSourceReferenceEntry => dataSourceReferenceEntry.Key, dataSourceReferenceEntry => dataSourceReferenceEntry.Value);
```

Comment:
`dataSourceReferences.Where(dataSourceReferenceEntry => dataSourceReferenceEntry.Value.dataSources.Count > 0)` is an IEnumerable<KeyValuePair<string, LDRJson>> where we've already ensured uniqueness of keys

Instead of using .ToDictionary, just pass it to a dictionary constructor. Avoids extra allocations for LINQ

Reply:
why not ` app._dataSourceReferences = new Dictionary<.., .. > (dataSourceReferences.Where(dataSourceReferenceEntry => dataSourceReferenceEntry.Value.dataSources.Count > 0))`

Reply:
I may be misunderstanding it here but i couldn't find overload for Dictionary constructor that takes in IEnumerable<KeyValuePair<...>>. I changed the code to avoid extra allocations of KeyValuePair objects. I need to filter out keyvaluepairs with empty data sources from dataSourceReferences local var and move those to app.dataSourceReferences.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.-ctor?view=netcore-3.1#system-collections-generic-dictionary-2-ctor(system-collections-generic-ienumerable((system-collections-generic-keyvaluepair((-0-1)))))

Reply:
Target framework for Microsoft.PowerPlatform.Formulas.Tools is netstandard2.0. "Applies To" table for this overloaded constructor only mentions .NET standard 2.1. I only get 6 overloaded constructor options and I get compiler error when I try to pass in the Ienumerable of keyvalue pairs as an argument to dictionary constructor. Not sure if something is wrong with my local setup

Reply:
Ah, nope. My bad! I didn't check that it wasn't in Standard 2.0. Feel free to ignore this.

Reply:
No worries! It's good to know that this exists. Didn't know about this overloaded constructor. Would have used Linq in other places as well. Thanks for making me aware of this.